### PR TITLE
[vaultwarden] Rewrite, Using pre-built binaries

### LINKED ISSUE #708: [vaultwarden] Bitwarden guide would only work with dev/nightly builds of Rust
The bitwarden guide currently fails at this point:
```
[bitwtest@stardust bitwarden_rs]$ cargo build -j 1 --release --features sqlite
[...]
error: failed to run custom build command for `pear_codegen v0.1.4`

Caused by:
  process didn't exit successfully: `/home/bitwtest/bitwarden_rs/target/release/build/pear_codegen-eb17718987529976/build-script-build` (exit code: 101)
--- stderr
Error: Pear requires a 'dev' or 'nightly' version of rustc.
Installed version: 1.43.1 (2020-05-04)
Minimum required:  1.31.0-nightly (2018-10-05)
thread 'main' panicked at 'Aborting compilation due to incompatible compiler.', /home/bitwtest/.cargo/registry/src/github.com-1ecc6299db9ec823/pear_codegen-0.1.4/build.rs:24:13
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

There is an issue about that: [Pear requires a nightly or dev version of Rust](https://github.com/SergioBenitez/Pear/issues/23). The author of the package in question states:

> The message continues to be correct for the time being and will disappear in a later release of pear that doesn't require nightly.

Uberspace is providing a version of Rust that is even way more _recent_ than the minimum required version, but it is a stable version, not a dev or nightly version, so the check of Pear fails.

I do not see any reason why Uberspace should provide a nightly build if a current stable build is available. So until this is fixed in the Pear module, I would suggest to - temporarily - drop the guide because I do not see how it could be made working.

### LINKED ISSUE #1160: [vaultwarden] compiler is killed during install/upgrade
see https://twitter.com/cblte/status/1470153178155044866

### LINKED ISSUE #1164: [vaultwarden] update section needs to be improved
It looks like there is a problem while upgrading and compiling vaultwarden with a newer rustc version than used in the initial install. It looks like the info in `~/vaultwarden/target/.rustc_info.json` will be outdated then. There could be other problems with cached files, so we should check how to to a fresh update process here.

Moreover, there is a problem with the current folder names shown in the guide:
image

also the line `wget new-release.tar.gz` should be improved to show better how to download the new source

### PR DESCRIPTION
**Pull request by knhash**
Resolves #1164, resolves  #1160, and resolves #708

### CODE REVIEW

**Comment by knhash** [comment]
Should close the following issues: https://github.com/Uberspace/lab/issues/1164, https://github.com/Uberspace/lab/issues/1160, https://github.com/Uberspace/lab/issues/708

**Comment by PapaBravo** [comment]
Followed the guide and it worked. Thanks 👍

**Comment by jce-zz** [comment]
Finally, this also worked for me! The normal build process does not work anymore as I assume the compiler runs out of memory and gets killed! :-(

**Comment by noave** [comment]
thanks for this! I added some comments, if you still could solve them I would merge this.

**Comment by noave** [comment]
> Hey @noave , I don't see the comments...

oops sorry, didn't send the review, but now there you are :)

File: source/guide_bitwarden.rst [hunk 1]
```diff
@@ -60,82 +52,55 @@ Installation
...
-
-In order to build vaultwarden successfully you'll need to set an environment variable pointing to the sqlite3 header files:
-
-.. code-block:: console
-
- [isabell@stardust ~]$ export SQLITE3_LIB_DIR=/var/lib64
+We will be installing vaultwarden by extracting a standalone, statically-linked binary from the official Docker image.
 
-``cd`` into the automatically created folder and create the ``data`` directory:
+Create a directory in ``/home/isabell`` for vaultwarden and its files.
```
[the remark this answers is not in the document]
Comment (knhash) [reply source/guide_bitwarden.rst:57]: Should be "vaultwarden" according to official guides. Changed accordingly.

File: source/guide_bitwarden.rst [hunk 2]
```diff
@@ -182,21 +162,22 @@ Use your favourite editor to create ``~/etc/services.d/vaultwarden.ini`` with th
...
- [program:vaultwarden]
- directory=%(ENV_HOME)s/vaultwarden
- command=%(ENV_HOME)s/vaultwarden/target/release/vaultwarden
- autostart=yes
- autorestart=yes
+  [program:vaultwarden]
+  directory=%(ENV_HOME)s/vaultwarden/output/
+  command=%(ENV_HOME)s/vaultwarden/output/vaultwarden
+  autostart=yes
+  autorestart=yes
```
[the remark this answers is not in the document]
Comment (knhash) [reply source/guide_bitwarden.rst:169]: Confirmed is enough. Added param.

**Pull request merged**